src/snowflake/snowpark/dataframe_na_functions.py

```diff
@@ -392,7 +392,8 @@ def fill(
         if not value_dict:
             new_df = copy.copy(self._dataframe)
             add_api_call(new_df, "DataFrameNaFunctions.fill")
-            new_df._ast_id = stmt.var_id.bitfield1
+            if _emit_ast:
+                new_df._ast_id = stmt.var_id.bitfield1
```

Comment:
Intuitively, this pattern seems suspicious. Is it OK for two distinct object to share the same `VarId`?

Reply:
that's a good question, here I basically simply assigned (like in other APIs) the id of the stmt to the new dataframe returned. The if stmt is needed, as the code is mixed (in some APIs we have a clear separation of AST code vs. existing, in others it is mixed in because of some parameter processing).